File: crates/ecp-cli/tests/fixtures/incremental_parity/ruby/user_service.rb
require_relative 'user'

class UserService
  def initialize(repository)
    @repository = repository
  end

  def find_by_id(id)
    @repository.find_by_id(id)
  end

  def find_all
    @repository.find_all
  end

  def create(email, name)
    user = User.new(nil, email, name)
    @repository.save(user)
  end

  def delete(id)
    @repository.delete(id)
  end
end

File: crates/ecp-cli/tests/fixtures/incremental_parity/csharp/UserService.cs
using System.Collections.Generic;
using System.Linq;

namespace Example
{
    public class UserService
    {
        private readonly IUserRepository _repository;

        public UserService(IUserRepository repository)
        {
            _repository = repository;
        }

        public User GetById(long id) => _repository.FindById(id);

        public IEnumerable<User> GetAll() => _repository.FindAll();

        public User Create(string email, string name) =>
            _repository.Save(new User { Email = email, Name = name });

        public void Delete(long id) => _repository.Delete(id);
    }
}
